src/hooks/useDrawer/useDrawer.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback, useState } from 'react'
+
+import { DrawerAPI } from './types'
+
+/**
+ * useDrawer hook allows for hiding visibility state .
+ *
+ * @returns {DrawerAPI} An object which includes the Drawer component, the drawer state (visible or hidden) and methods
```

Comment:
The returned object does not include the component. We should either change the doc or add a memorized component to the hook (as in the `useModal` hook). I would go for the latter, otherwise this hook seems a bit redundant

Reply:
>The returned object does not include the component. 

this is sadly intended, I've started developing the useDrawer hook as a mean of hiding the visible state **and** keeping it associated with a component (as the useModal does); however there is a problem with the visibility state management within the hook that causes the re-creation of the component and the lack of animations during the drawer close.

For this first implementation I decided not to include the component in the hook result as it would produce a buggy drawer, we should investigate it in the future to make the useDrawer hook more useful since, at least it still hides the state management part...

I'm changing the doc here